inkscape/src/main/assembly/dist/lace_polar.py

```diff
@@ -134,50 +137,36 @@ def generate(self):
 				diameter += self.iterate(diameter, circleNr)
 				circleNr += 1
 
-	def variantRectangle(self):
+	def removeGroups(self, start, increment):
 		"""
-		Remove dots 
+		Remove complete rings with dots
 		"""
-		i = 1
-		while (i < self.nrOfGeneratedCircles):
+		for i in range(start, self.nrOfGeneratedCircles, increment):
 			self.current_layer.remove(self.generatedCircles[i])
-			i += 2
-
-	def variantHexagon1(self):
-		"""
-		Remove dots 
-		"""
-		i = 2
-		while (i < self.nrOfGeneratedCircles):
-			self.current_layer.remove(self.generatedCircles[i])
-			i += 3
 
 	def variantHexagon2(self):
 		"""
 		Remove dots 
 		"""
-		i = 1
-		while (i < self.nrOfGeneratedCircles):
+		for i in range(1, self.nrOfGeneratedCircles, 1):
 			j = 0
-			for dot in self.generatedCircles[i].iterchildren():
-				if ((((i+1) % 2) + j) % 3) == 0 :
-					self.generatedCircles[i].remove(dot)
+			group = self.generatedCircles[i]
+			for dot in group.iterchildren():
+				if ((((i + 1) % 2) + j) % 3) == 0 :
+					group.remove(dot)
 				j += 1
-			i += 1
 
 	def variantHexagon3(self):
 		"""
 		Remove dots 
 		"""
-		i = 1
-		while (i < self.nrOfGeneratedCircles):
-			if ((i%2) == 1 ):
-				j = 0
-				for dot in self.generatedCircles[i].iterchildren():
-					if (((((i+1) % 4)/2) + j) % 2) == 0 :
-						self.generatedCircles[i].remove(dot)
-					j += 1
-			i += 1
+		for i in range(2, self.nrOfGeneratedCircles, 2):
+			j = 0
+			group = self.generatedCircles[i]

```

Comment:
Tried another type of loop in my polar-loops branch, but "list(group)[j]" seems to render only half of the dots. https://github.com/jo-pol/DiBL/blob/polar-loops/inkscape/src/main/assembly/dist/lace_polar.py#L164